### PR TITLE
Bigflow starter

### PR DESCRIPTION
**Pull request by iwko**

### CODE REVIEW

**Comment by beczkowb** [comment]
About the dialog in the `start-project` command:

1. Give more description for each step. For example, instead of "GCP bucket", say something like: "Cloud Composer bucket id - used to automatically deploy your workflows to a Cloud Composer"
2. Show the available options if possible. For example, when you ask "Would you like to create basic or advanced project?", I don't know if I should write "basic" in response or something else. Another case: when you ask for a GCP project, you could list the available projects using `gcloud projects list`.
3. Provide a reasonable default if possible.
4. Put a link to the documentation that corresponds to a question (so user knows where to find the details).
5. Add some welcome message and asci logo, I think it's a nice detail.

File: bigflow/cli.py [hunk 1]
```diff
@@ -236,12 +237,18 @@ def _parse_args(project_name: Optional[str], args) -> Namespace:
...
+    _create_start_project_parser(subparsers)
+
     _create_project_version_parser(subparsers)
     _create_release_parser(subparsers)
 
     return parser.parse_args(args)
 
 
+def _create_start_project_parser(subparsers):
+    subparsers.add_parser('start-project', description='Creates example project in current directory.')
```
Comment (bartoszwalacik) [inline bigflow/cli.py:249]: Creates a scaffolding project in a current directory.

File: bigflow/cli.py [hunk 2]
```diff
@@ -545,6 +552,44 @@ def _cli_build(args):
...
+def gcp_bucket_input():
+    return input('GCP Bucket\n')
+
+
+def environment_name_input():
+    return input('Environment name\n')
+
+
+def project_name_input():
+    return input('Project name\n')
```
Comment (bartoszwalacik) [inline bigflow/cli.py:576]: Project name -> Enter the project name:

File: bigflow/cli.py [hunk 3]
```diff
@@ -545,6 +552,44 @@ def _cli_build(args):
     run_process(cmd)
 
 
+def project_type_input():
+    return input("Would you like to create basic or advanced project?\n")
```
Comment (bartoszwalacik) [inline bigflow/cli.py:556]: I don't know what to type - (y/n)?

Comment (bartoszwalacik) [inline bigflow/cli.py:556]: we need validation here, now there is no validation:

```
Project name
bgq-22
Would you like to create basic or advanced project?
fddvc
How many GCP projects would you like to use?


```

File: bigflow/cli.py [hunk 4]
```diff
@@ -545,6 +552,44 @@ def _cli_build(args):
...
+
+
+def project_number_input():
+    return input('How many GCP projects would you like to use?\n')
```
Comment (bartoszwalacik) [inline bigflow/cli.py:560]: This question is unclear. First, why it's important, second, give me options

File: bigflow/cli.py [hunk 5]
```diff
@@ -545,6 +552,44 @@ def _cli_build(args):
...
+
+
+def gcp_project_input():
+    return input('GCP project\n')
```
Comment (bartoszwalacik) [inline bigflow/cli.py:564]: which one if I answered `2` to the previous question

Comment (bartoszwalacik) [inline bigflow/cli.py:564]: for each question, I expect default answer,  which I would change or accept

File: bigflow/cli.py [hunk 6]
```diff
@@ -545,6 +552,44 @@ def _cli_build(args):
...
+
+
+def gcp_bucket_input():
+    return input('GCP Bucket\n')
```
Comment (bartoszwalacik) [inline bigflow/cli.py:568]: GCP Bucket -> GCP Storage Bucket which underlies your Composer's DAGs Folder

File: bigflow/cli.py [hunk 7]
```diff
@@ -545,6 +552,44 @@ def _cli_build(args):
...
+
+
+def environment_name_input():
+    return input('Environment name\n')
```
Comment (bartoszwalacik) [inline bigflow/cli.py:572]: I think that interactively defining more than one env  is too complex

**Comment by bartoszwalacik** [comment]
Interakcja z narzędziem urywa się nagle po odpowiedzi na ostanie pytanie, nie wiem co się stało, powinien być jakiś komunikat na koniec, typu: hej właśnie wygenerowałeś swój projekt, projekt jest tu i tu, jest sukces

File: bigflow/scaffold/scaffold_templates.py [hunk 8]
```diff
@@ -0,0 +1,348 @@
+readme_template = '''# Run project
```
Comment (bartoszwalacik) [inline bigflow/scaffold/scaffold_templates.py:1]: Why Run project as H1 ?

**Comment by beczkowb** [comment]
The starter produces `workflows` package. It should be called as the project name. So for example, when I create "datacraft" project, I want to have the main package called "datacraft" (not "workflows").

File: bigflow/scaffold/scaffold_templates.py [hunk 9]
```diff
@@ -0,0 +1,348 @@
...
+
+### Requirements
+  1. [GCP SDK](https://cloud.google.com/sdk/docs).
+
+### Preparations
+  1. Log in using SDK (`gcloud auth application-default login`).
```
Comment (bartoszwalacik) [inline bigflow/scaffold/scaffold_templates.py:7]: Log in to GCP using Google Cloud SDK

**Comment by beczkowb** [comment]
The generated requirements look like this:
```
apache-beam==2.23.0
google-apitools==0.5.31
wheel==0.35.1
```

Why beam has no gcp addon? Why do we need api-tools and wheel?

File: bigflow/scaffold/scaffold_templates.py (hunk 9 above)
Comment (bartoszwalacik) [inline bigflow/scaffold/scaffold_templates.py:7]: commands should not be standalone not inlined, so :

Log in to GCP using Google Cloud SDK:

```
gcloud auth application-default login
```

File: bigflow/scaffold/scaffold_templates.py [hunk 10]
```diff
@@ -0,0 +1,348 @@
...
+  1. You need to create [Cloud Composer] (https://cloud.google.com/composer/docs/how-to/managing/creating#creating_a_new_environment).
```
Comment (bartoszwalacik) [inline bigflow/scaffold/scaffold_templates.py:8]: You need to create [Cloud Composer]
Create a [Cloud Composer] instance

File: bigflow/scaffold/scaffold_templates.py [hunk 11]
```diff
@@ -0,0 +1,348 @@
...
+  1. You need to create GCS bucket `{project_name}` and inside this bucket create directories `beam_runner/staging` and `beam_runner/temp`.
+  1. In your deployment_config change key docker_repository to point your docker registry (we recommend to use one docker repository for all environments).
```
Comment (bartoszwalacik) [inline bigflow/scaffold/scaffold_templates.py:10]: Imo ten opis jest zbyt szczegółowy tu, pomyślcie jak trudno będzie potem to zmienić/poprawić w już wygenerowanych plikacj. Jeśli to jest instrukcja, to przenieśmy ją do docsów i podlinkujmy tu tylko

File: bigflow/scaffold/scaffold_templates.py [hunk 12]
```diff
@@ -0,0 +1,348 @@
...
+  1. In bucket connected to your docker registry you need to add role Storage Object Admin pointing to Service Account that composer uses.
+  1. Install all required dependencies `pip install -r resources/requirements.txt`
+
+### Running locally
+  1. To run bigquery workflow type in project directory `bf run --workflow internationalports`.
```
Comment (bartoszwalacik) [inline bigflow/scaffold/scaffold_templates.py:15]: one example is enough:

To run a workflow type:

```
bigflow run --workflow internationalports
```

File: bigflow/scaffold/scaffold_templates.py [hunk 13]
```diff
@@ -0,0 +1,348 @@
...
+  1. To run apache beam workflow on Dataflow type in project directory `bf run --workflow wordcount`.
+  
+### Building & deploying
+  1. You need to build docker image of your project and DAG file for your composer. To do this type in terminal `bf build -e`.
+  1. When build is completed in /image directory you can find created docker image and in .dags directory you can find
```
Comment (bartoszwalacik) [inline bigflow/scaffold/scaffold_templates.py:20]: im mniej tekstu tu tym lepiej, raz że już to gdzie indziej opisujemy a dwa że zaraz się zdezaktualizuje , np nie `.dags` tylko  `dags`

tu powinna być tylko opisane krótko podstawowe komendy z linkami do dokumentacji, np

[Build](..) the project:

```
bigflow build   
```

[Deploy](,,,) the project to Composer: 

```
bigflow deploy
```

[Run](...) a workflow on given env:

```
bigflow ...
```

**Review by bartoszwalacik — CHANGES_REQUESTED** [review]

**Comment by beczkowb** [comment]
Take care of the formatting. For example:

Zrzut ekranu 2020-09-3 o 13 54 09

**Comment by beczkowb** [comment]
Instead of:
```
options.view_as(SetupOptions).setup_file = resolve(
        create_file_if_not_exists(find_file('workflows', Path(__file__)).parent / 'setup.py', create_setup_body('workflows')))
```

Use:
```
    options.view_as(SetupOptions).setup_file = resolve(find_or_create_setup_for_main_project_package('workflows', Path(__file__)))
```

**Comment by beczkowb** [comment]
Instead of:
```
from setuptools import setup
from bigflow.build import project_setup, auto_configuration

PROJECT_NAME = 'workflows'

if __name__ == '__main__':
    setup(**project_setup(**auto_configuration(PROJECT_NAME)))
```

Use:
```
from bigflow.build import default_project_setup

PROJECT_NAME = 'workflows'

if __name__ == '__main__':
    default_project_setup(PROJECT_NAME)
```

**Comment by iwko** [comment]
> The generated requirements look like this:
> 
> ```
> apache-beam==2.23.0
> google-apitools==0.5.31
> wheel==0.35.1
> ```
> 
> Why beam has no gcp addon? Why do we need api-tools and wheel?

beam is required for tests here. We need wheel for `bf build` command but maybe it will be better idea to move it to bigflo. We need api-tools in composer.

File: bigflow/scaffold/scaffold_templates.py [hunk 14]
```diff
@@ -0,0 +1,348 @@
+readme_template = '''# Run project
+
+### Requirements
+  1. [GCP SDK](https://cloud.google.com/sdk/docs).
```
[the remark this answers is not in the document]
Comment (bartoszwalacik) [reply bigflow/scaffold/scaffold_templates.py:4]: najlepiej w ogóle nie opisuj jak instalować bf tylko zalinkuj do tej sekcji dokumentacji [Installing BigFlow](https://github.com/allegro/bigflow/blob/master/README.md#installing-bigflow) 
będzie taka

**Comment by beczkowb** [comment]
I would like to also get reasonable .gitignore

File: bigflow/scaffold/scaffold_templates.py (hunk 12 above)
Comment (iwko) [reply bigflow/scaffold/scaffold_templates.py]: I want to show here that there are two separate workflows

**Comment by iwko** [comment]
@beczkowb requirements solved. wheel is needed in scaffold_project, apitools moved to bigflow dependencies

**Comment by beczkowb** [comment]
"Enter the project name. It will be used as a directory and dataflow bucket name."

1. You should mention that the provided project name should be a valid python package name. 
2. As a directory for what? dataflow bucket name for what? It's too generic.
3. Give links to details.

**Comment by beczkowb** [comment]
"Enter a GCP project that would you use. Above is a list of available projects. If not provided default project from gcloud will be used."

1. Print the default project that you mention here - "If not provided default project from gcloud will be used"
2. Give me more information, why exactly do you need a gcp project id?
3. "Enter a GCP project that would you use" -> "Enter a GCP project ID that you want to use in your project", something like that.
4. "Above is a list of available projects" -> "Choose a project from the list above"

**Comment by beczkowb** [comment]
"Enter a Cloud Composer Bucket name where DAG files will be stored."

Give me links where I can find more information about cloud composer, dags, dags buckets.

**Comment by beczkowb** [comment]
Maybe instead of:
```
from bigflow.bigquery import DatasetConfig

INTERNAL_TABLES = ['ports', 'more_ports']

EXTERNAL_TABLES = {}

dataset_config = DatasetConfig(env='dev',
                               project_id='sc-11309-content-scorer-dev',
                               dataset_name='bigflow_test',
                               internal_tables=INTERNAL_TABLES,
                               external_tables=EXTERNAL_TABLES
                               )
```

Like this:
```
from bigflow.bigquery import DatasetConfig

dataset_config = DatasetConfig(
    env='dev',
    project_id='sc-11309-content-scorer-dev',
    dataset_name='bigflow_test',
    internal_tables=['ports', 'more_ports'])
```

**Comment by beczkowb** [comment]
Use the same formatting for all the SQLs you generate.

**Comment by beczkowb** [comment]
Why do you put SQL code to variables and then you use the variables in the components?

**Comment by beczkowb** [comment]
```
bigflow[bigquery]==1.0.dev30
apache-beam==2.23.0
wheel==0.35.1
```

Why do we need wheel? Maybe you should put it into the bigflow requirements?

**Comment by beczkowb** [comment]
```
create_tables_job = create_tables.to_job(id=None, dependencies_override={'dataset': dataset})
ports_job = ports.to_job(id=None, dependencies_override={'dataset': dataset})
populate_job = populate_more_ports.to_job(id=None, dependencies_override={'dataset': dataset})

ports_workflow = Workflow(
        workflow_id='test_bigquery_workflow',
        definition=[create_tables_job, populate_job, ports_job],
        schedule_interval='@once')
```

Why `id` as None?

`test_bigquery_workflow` -> `internationalports`

**Comment by beczkowb** [comment]
```
from bigflow.configuration import Config

workflow_config = Config(name='dev',
                         properties={
                             'project_name': 'datacraftacademy',
                             'gcp_project_id': 'sc-11309-content-scorer-dev',
                             'staging_location': 'datacraftacademy/beam_runner/staging',
                             'temp_location': 'datacraftacademy/beam_runner/temp',
                             'region': 'europe-west1',
                             'machine_type': 'n1-standard-1'
                         }).resolve()
```

```
from bigflow.configuration import Config

workflow_config = Config(
    name='dev',
    properties={
        'project_name': 'datacraftacademy',
        'gcp_project_id': 'sc-11309-content-scorer-dev',
        'staging_location': 'datacraftacademy/beam_runner/staging',
        'temp_location': 'datacraftacademy/beam_runner/temp',
        'region': 'europe-west1',
        'machine_type': 'n1-standard-1'
    }).resolve()
```

**Comment by beczkowb** [comment]
```
def dataflow_pipeline(gcp_project_id, staging_location, temp_location, region, machine_type, project_name):
    import uuid
    from pathlib import Path

    import apache_beam as beam

    from apache_beam.options.pipeline_options import SetupOptions, StandardOptions, WorkerOptions, GoogleCloudOptions, PipelineOptions

    from bigflow.resources import find_or_create_setup_for_main_project_package, resolve, get_resource_absolute_path
    options = PipelineOptions()

    google_cloud_options = options.view_as(GoogleCloudOptions)
    google_cloud_options.project = gcp_project_id
    google_cloud_options.job_name = f'beam-wordcount-{uuid.uuid4()}'
    google_cloud_options.staging_location = f"gs://{staging_location}"
    google_cloud_options.temp_location = f"gs://{temp_location}"
    google_cloud_options.region = region

    options.view_as(WorkerOptions).machine_type = machine_type
    options.view_as(WorkerOptions).max_num_workers = 2
    options.view_as(WorkerOptions).autoscaling_algorithm = 'THROUGHPUT_BASED'
    options.view_as(StandardOptions).runner = 'DataflowRunner'

    options.view_as(SetupOptions).setup_file = resolve(find_or_create_setup_for_main_project_package(project_name, Path(__file__)))
    options.view_as(SetupOptions).requirements_file = resolve(get_resource_absolute_path('requirements.txt', Path(__file__)))
    return beam.Pipeline(options=options)
```

```
import uuid
from pathlib import Path

import apache_beam as beam
from apache_beam.options.pipeline_options import SetupOptions, StandardOptions, WorkerOptions, GoogleCloudOptions, PipelineOptions
    
    
def dataflow_pipeline(gcp_project_id, staging_location, temp_location, region, machine_type, project_name):
    from bigflow.resources import find_or_create_setup_for_main_project_package, resolve, get_resource_absolute_path

    options = PipelineOptions()

    google_cloud_options = options.view_as(GoogleCloudOptions)
    google_cloud_options.project = gcp_project_id
    google_cloud_options.job_name = f'beam-wordcount-{uuid.uuid4()}'
    google_cloud_options.staging_location = f"gs://{staging_location}"
    google_cloud_options.temp_location = f"gs://{temp_location}"
    google_cloud_options.region = region

    options.view_as(WorkerOptions).machine_type = machine_type
    options.view_as(WorkerOptions).max_num_workers = 1
    options.view_as(StandardOptions).runner = 'DataflowRunner'

    options.view_as(SetupOptions).setup_file = resolve(find_or_create_setup_for_main_project_package(project_name, Path(__file__)))
    return beam.Pipeline(options=options)
```

**Comment by beczkowb** [comment]
wordcount.workflow:

```
from apache_beam.io import WriteToText
from bigflow import Workflow

from .pipeline import dataflow_pipeline
from .config import workflow_config
from .processing import count_words


class WordcountJob(object):
    def __init__(self, gcp_project_id, staging_location, temp_location, region, machine_type, project_name):
        self.id = 'count_words'
        self.retry_count = 20
        self.retry_pause_sec = 100

        self.gcp_project_id = gcp_project_id
        self.staging_location = staging_location
        self.temp_location = temp_location
        self.region = region
        self.machine_type = machine_type
        self.project_name = project_name

    def run(self, runtime):
        p = dataflow_pipeline(self.gcp_project_id, self.staging_location, self.temp_location,
                              self.region, self.machine_type, self.project_name)
        count_words(p, WriteToText("gs://{}/beam_wordcount".format(self.temp_location)))
        p.run().wait_until_finish()


wordcount_workflow = Workflow(
    workflow_id="wordcount",
    definition=[WordcountJob(
        gcp_project_id=workflow_config['gcp_project_id'],
        staging_location=workflow_config['staging_location'],
        temp_location=workflow_config['temp_location'],
        region=workflow_config['region'],
        machine_type=workflow_config['machine_type'],
        project_name=workflow_config['project_name']
    )])
```

**Comment by beczkowb** [comment]
Empty line

```

from bigflow.build import default_project_setup

PROJECT_NAME = 'datacraftacademy'

if __name__ == '__main__':
    default_project_setup(PROJECT_NAME)
```

**Comment by beczkowb** [comment]
Add `image` and `.dags` to the `.gitignore`

File: bigflow/scaffold/scaffold_templates.py [hunk 15]
```diff
@@ -0,0 +1,478 @@
+readme_template = '''## Run project
+
+### Requirements
```
Comment (beczkowb) [inline bigflow/scaffold/scaffold_templates.py:3]: Squish the `Requirements` and `Preparations` to a single section.

File: docs/scaffold.md [hunk 16]
```diff
@@ -0,0 +1,28 @@
+# Start project
+
+## Overview
+
+To see how easy is using Bigflow you can use `start-project` which creates example standalone project with [standard structure](https://github.com/allegro/bigflow/blob/master/docs/project_setup_and_build.md#project-structure).
```
Comment (ewebed) [inline docs/scaffold.md:5]: which creates a sample standalone project

File: docs/scaffold.md [hunk 17]
```diff
@@ -0,0 +1,28 @@
...
+This project contains Apache Beam workflow and BigQuery workflow.
```
Comment (ewebed) [inline docs/scaffold.md:6]: an Apache Beam workflow and a BigQuery workflow

File: docs/scaffold.md [hunk 18]
```diff
@@ -0,0 +1,28 @@
...
+You will be able to deploy those workflows in your GCP project providing informations like project_id/bucket etc.
+
+Beam workflow example counts letters in array and then saves this information to file in your bucket.
```
Comment (ewebed) [inline docs/scaffold.md:9]: The Beam workflow example counts letters in an array and then saves this information to a file in your bucket.

File: docs/scaffold.md [hunk 19]
```diff
@@ -0,0 +1,28 @@
...
+BigQuery workflow example creates necessary tables then populates one of them and finally moves data from one table to another.
```
Comment (ewebed) [inline docs/scaffold.md:10]: The BigQuery workflow example

File: docs/scaffold.md [hunk 20]
```diff
@@ -0,0 +1,28 @@
...
+
+
+## Start project
+To start a new project type in terminal
```
Comment (ewebed) [inline docs/scaffold.md:14]: To start a new project, type in a terminal

File: docs/cli.md [hunk 21]
```diff
@@ -466,4 +466,7 @@ bigflow deploy \
 --dags-bucket europe-west1-12323a-bucket \
 --gcp-project-id my_gcp_dev_project \
 --clear-dags-folder
-```
\ No newline at end of file
+```
+
+## Scaffold project
+Use `bigflow start-project` command to create [example project](scaffold.md) and try all of the above commands by yourself.
```
Comment (ewebed) [inline docs/cli.md:472]: usa a `bigflow start-project` command to create a [sample project](scaffold.md) and try all of the above commands yourself.

File: docs/scaffold.md [hunk 22]
```diff
@@ -0,0 +1,28 @@
...
+BigQuery workflow example creates necessary tables then populates one of them and finally moves data from one table to another.
+
+
+## Start project
+To start a new project type in terminal 
+
+    bigflow start-project 
+
+## Preparation
+  1. Log in to GCP using Google Cloud SDK:
```
Comment (ewebed) [inline docs/scaffold.md:19]: Log in to (the) GCP, using the Google Cloud SDK:

File: docs/scaffold.md [hunk 23]
```diff
@@ -0,0 +1,28 @@
...
+  
+    gcloud auth application-default login.
```
Comment (ewebed) [inline docs/scaffold.md:21]: to nie powinno być w jakichś ciapkach?

File: docs/scaffold.md [hunk 24]
```diff
@@ -0,0 +1,28 @@
...
+    
+  1. Create a [Cloud Composer](https://cloud.google.com/composer/docs/how-to/managing/creating#creating_a_new_environment) instance.
+  1. You need to create GCS bucket with same name as your project and inside this bucket create directories `beam_runner/staging` and `beam_runner/temp`. This bucket is for dataflow processing.
```
Comment (ewebed) [inline docs/scaffold.md:24]: 1. You need to create a GCS bucket with the same name as your project and inside this bucket create directories `beam_runner/staging` and `beam_runner/temp`. 

This bucket is for dataflow processing. <- to ostatnie zdanie brzmi tak średnio, ale nie wiem jak zmienić :(

File: docs/scaffold.md [hunk 25]
```diff
@@ -0,0 +1,28 @@
...
+  1. In your deployment_config change key docker_repository to point your Docker registry (we recommend to use one Docker repository for all environments).
```
Comment (ewebed) [inline docs/scaffold.md:25]: In your deployment_config change the docker_repository key to point to your Docker registry (we recommend to use one Docker repository for all environments).

File: docs/scaffold.md [hunk 26]
```diff
@@ -0,0 +1,28 @@
...
+  1. In bucket connected to your Docker registry you need to add role Storage Object Admin pointing to Service Account that Composer uses.
```
Comment (ewebed) [inline docs/scaffold.md:26]: In the bucket connected to your Docker registry, you need to add a role "Storage Object Admin" pointing to the Service Account that the Composer uses.

File: docs/scaffold.md [hunk 27]
```diff
@@ -0,0 +1,28 @@
...
+You will be able to deploy those workflows in your GCP project providing informations like project_id/bucket etc.
+
+Beam workflow example counts letters in array and then saves this information to file in your bucket.
+BigQuery workflow example creates necessary tables then populates one of them and finally moves data from one table to another.
+
+
+## Start project
+To start a new project type in terminal 
+
+    bigflow start-project
```
Comment (beczkowb) [inline docs/scaffold.md:16]: Why are you not using code markdown syntax and tag instead?

File: docs/scaffold.md (hunk 23 above)
Comment (beczkowb) [inline docs/scaffold.md:21]: same here, why not code

File: docs/scaffold.md (hunk 22 above)
Comment (beczkowb) [inline docs/scaffold.md:19]: I think you can link to the BigFlow installation guide

File: docs/scaffold.md (hunk 24 above)
Comment (beczkowb) [inline docs/scaffold.md:24]: Provide some link from beam documentation if possible

File: docs/scaffold.md (hunk 25 above)
Comment (beczkowb) [inline docs/scaffold.md:25]: `deployment_config.py`

Comment (beczkowb) [reply docs/scaffold.md:25]: `docker_repository`

File: docs/scaffold.md [hunk 28]
```diff
@@ -0,0 +1,28 @@
...
+Beam workflow example counts letters in array and then saves this information to file in your bucket.
+BigQuery workflow example creates necessary tables then populates one of them and finally moves data from one table to another.
+
+
+## Start project
+To start a new project type in terminal 
+
+    bigflow start-project 
+
+## Preparation
```
Comment (beczkowb) [inline docs/scaffold.md:18]: I think that the deployment doc will describe most of the things which you describe here, @bartoszwalacik what do you think?

File: docs/scaffold.md (hunk 22 above)
Comment (iwko) [reply docs/scaffold.md:19]: Only for one step?

Comment (iwko) [reply docs/scaffold.md:19]: Why? There is no info about using application-default login

File: docs/scaffold.md [hunk 29]
```diff
@@ -0,0 +1,28 @@
...
+The BigQuery workflow example creates necessary tables then populates one of them and finally moves data from one table to another.
+
+
+## Start project
+To start a new project, type in terminal 
+
+    bigflow start-project 
+
+## Preparation
+  1. Log in to the, GCP using Google Cloud SDK:
```
Comment (ewebed) [inline docs/scaffold.md:19]: Log in to the GCP, using Google Cloud SDK:

File: docs/cli.md [hunk 30]
```diff
@@ -466,4 +466,7 @@ bigflow deploy \
 --dags-bucket europe-west1-12323a-bucket \
 --gcp-project-id my_gcp_dev_project \
 --clear-dags-folder
-```
\ No newline at end of file
+```
+
+## Scaffold project
+Use a `bigflow start-project` command to create [sample project](scaffold.md) and try all of the above commands yourself.
```
Comment (ewebed) [inline docs/cli.md:472]: to create a [sample project]

File: docs/scaffold.md [hunk 31]
```diff
@@ -0,0 +1,28 @@
+# Start project
+
+## Overview
+
+To see how easy it is to use Bigflow you can use `start-project` which creates a sample standalone project with a [standard structure](https://github.com/allegro/bigflow/blob/master/docs/project_setup_and_build.md#project-structure).
+This project contains an Apache Beam workflow and a BigQuery workflow.
+You will be able to deploy those workflows in your GCP project providing informations like project_id/bucket etc.
```
Comment (ewebed) [inline docs/scaffold.md:7]: You will be able to deploy those workflows on your GCP project providing information like project_id/bucket etc.

File: docs/scaffold.md [hunk 32]
```diff
@@ -0,0 +1,28 @@
...
+
+    bigflow start-project 
+
+## Preparation
+  1. Log in to the, GCP using Google Cloud SDK:
+  
+    gcloud auth application-default login.
+    
+  1. Create a [Cloud Composer](https://cloud.google.com/composer/docs/how-to/managing/creating#creating_a_new_environment) instance.
+  1. You need to create aGCS bucket with the same name as your project and inside this bucket create directories `beam_runner/staging` and `beam_runner/temp`. [Dataflow documentation](https://cloud.google.com/dataflow/docs/guides/specifying-exec-params#configuring-pipelineoptions-for-execution-on-the-cloud-dataflow-service)
```
Comment (ewebed) [inline docs/scaffold.md:24]: a GCP

File: docs/scaffold.md [hunk 33]
```diff
@@ -0,0 +1,28 @@
...
+  1. In your `deployment_config.py` change the docker_repository key to point your Docker registry (we recommend to use one Docker repository for all environments).
```
Comment (ewebed) [inline docs/scaffold.md:25]: to point to your Docker

File: docs/scaffold.md (hunk 32 above)
Comment (iwko) [reply docs/scaffold.md:24]: GCS - google cloud storage

**Pull request merged**